zonemaster/zonemaster-gui
# Make it possible to translate buttons' tooltips/bubbles

## Purpose

The button tooltips are not translated. This makes it possible to use the translation file for each language.

## Context

Addresses #241 

## Changes

#### Main change
* support for tooltip translation using the `ngx-translate` and `ngx-extract` tools
* translate into French, Swedish (so far)

#### Side changes
* update the way we translate the "Disable IPv4" and "Disable IPv6" texts
* update the npm `i18n:extract` command to use 4 spaces instead of tabulations
* update the npm command to use _null_ value for unset translations
* run the `npm i18n:extract` command: this has sortes alphabetically each LANG.json file and replaced the tabs with 4 spaces
* reduce the length of the _copyright_ key

## How to test this PR

Set the GUI into English, then hover on the "All", "Info", "Notice", "Warning", "Error" and "Critical" buttons, make sure that there is a text for each tooltip (this should be the same as the one before the PR).
(if translations have been added within this PR, change the GUI language and do the same, the tooltip's text should be updated accordingly)

Side test, make sure that the "Disable IPv4" and "Disable IPv6" are correctly translated.


#### TODO

- [ ] translate into Danish
- [ ] translate into Finnish
- [x] translate into French
- [ ] translate into Norwegian
- [x] translate into Swedish
